codex-rs/core/src/config.rs

```diff
@@ -340,6 +343,129 @@ pub fn set_project_trusted(codex_home: &Path, project_path: &Path) -> anyhow::Re
     Ok(())
 }
 
+/// Persist the default `model` and `model_reasoning_effort` to
```

Comment:
Since this file is already quite large, can we consolidate logic related to config editing in another file? Maybe `config_edit.rs`? (I expect we'll be adding more code like this.)